### PR TITLE
[469348]: Add magic functions

### PR DESCRIPTION
**Pull request by danzatt**
I have implemented Python magic functions for the Vector structure (including tests).

### CODE REVIEW

**Comment by danzatt** [comment]
Ok, comment added, commits squashed.

File: ossdev/__init__.py [hunk 1]
```diff
@@ -34,38 +35,50 @@ def __hash__(self):
...
 
     def __setitem__(self, key, value):
-        # TODO: implement
-        return None
+        self.d[key] = value
 
     def __cmp__(self, other):
-        # TODO: implement, -1 if self < other, 0 if self == other, 1 if self > other
-        return -1
+        return len(self) - len(other)
```
Comment (ph4r05) [inline ossdev/__init__.py:41]: This does not seem like a valid vector comparison

File: ossdev/__init__.py [hunk 2]
```diff
@@ -34,38 +35,50 @@ def __hash__(self):
...
+
+    def __eq__(self, other):
```
Comment (ph4r05) [inline ossdev/__init__.py:43]: Implementing `cmp` should suffice, implementations of other comparators seems redundant.

**Review by ph4r05 — DISMISSED** [review]

**Comment by ph4r05** [comment]
Commits are not signed, but I approve this PR so you don't have to fix PR step 2 then.

**Review by ph4r05 — APPROVED** [review]

**Comment by ph4r05** [comment]
Approved, HW done, closing

**Pull request closed without merging**